linkml/generators/pythongen.py

```diff
@@ -131,6 +131,14 @@ def gen_schema(self) -> str:
             if self.genmeta
             else "from linkml_runtime.linkml_model.meta import EnumDefinition, PermissibleValue, PvFormulaOptions\n"
         )
+        enum_pvs = []
+        for enum_name, enum_def in self.schema.enums.items():
+            for pv in enum_def.permissible_values:
+                enum_pvs.append(f"{enum_name}({pv})")
+        for class_name, class_def in self.schema.classes.items():
+            for slot in class_def.slots:
+                if self.schema.slots[slot].ifabsent in enum_pvs:
+                    enumimports = enumimports + "from dataclasses import field\n"
```

Comment:
If multiple classes/slots satisfy this condition, I think this would result in multiple identical imports?

Reply:
pydanticgen `Import` and `Imports` classes handle deduplication and sorting of imports (and are not pydantic specific) https://github.com/linkml/linkml/blob/68a880a458708c20cde6337b3493ee93a85c3ba8/linkml/generators/pydanticgen/template.py#L263

Reply:
I have used the `Import` and `Imports` classes as proposed by @sneakers-the-rat. They are really cool!

I have not changed the rest of the imports to keep this PR scoped. But IMO a future PR should take care of using those same classes all the imports in the generated python code.

Reply:
BTW @pkalita-lbl good catch!